### DISCUSSION TITLE
posts/latent-mean-centering/

### DISCUSSION THREAD

**Discussion by giscus** [mvuorre/mvuorre.github.io#30] [giscus]
# posts/latent-mean-centering/

Researchers studying longitudinal data routinely center their predictors to isolate between- and within-cluster contrasts (Enders and Tofighi 2007). This within-cluster centering is usually an easy data-manipulation step. However, centering variables on the observed means can bias the resulting estimates, a problem that is avoided with latent mean centering. In this entry, we study how to latent-mean-center variables in multilevel models using brms.

https://mvuorre.github.io/posts/latent-mean-centering/

**Comment by CraterDylan** [comment]
Hello! As I guess for many others, I was led to this post by trying to replicate McElreath's latent mean centering from his rethinking course with brms, and I'm grateful for this post, which is one of the few valuable resources.
I wanted to ask what you thought of [Ruben Arslan's post](https://rubenarslan.github.io/posts/2023-03-11-latent-group-mean-centering-revisited/) in which applying your approach to McElreath's example, or a gaussian link version, doesn't seem to work. If I have to share my opinion, I think the me() approach is a bit clear, if nothing else. Anyway, thanks.

**Reply by mvuorre** [reply]
Hi, thanks! I'm not really sure what is going on in that blog post (and not sure what a 'mundlak' is), but the code as discussed in the current version of my blog post does what is advertised (I don't think the `nlf()` thing works?).

**Reply by CraterDylan** [reply]
Mundlak is the person giving the name to a similar kind of mean centering, where instead of subtracting the mean you adjust for it. The [bonus](https://youtu.be/iwVqiiXYeC4?t=3299) at the end of McElreath's lecture is a nice resource.
Indeed, there's a difference in replicating your approach there: the latent mean (here alpha, there Xbar) is included in that `nlf()` instead.
Anyway, the claim there is that your example dataset might not be able to show the problem. Maybe you or I can try with the McElreath example, code [here](https://github.com/rmcelreath/stat_rethinking_2023/blob/main/scripts/12_bonus_mundlak.r).
I will try both approaches on my own simulation, once I fix the non-latent mean model.
P.S.: looking closer at all three implementations, yours seems closer to McElreath's than Arslan's, modeling the latent mean with a random intercept only. I guess they all do that, but the other one has it wrapped inside the `nlf()` thing.

**Reply by mvuorre** [reply]
So given how brms works if you're either adjusting for (mundlak I guess) or subtracting latent cluster characteristics, you need to do the funky data stacking as described in the post--the nlf() syntax won't work because you can't link parameters across different outcomes. I think for the same reason the mi() and me() approaches shouldn't work because you cannot link the missing thing across outcomes--but I don't at the moment understand all that's going on in the blog post you link to.

**Reply by CraterDylan** [reply]
That's ok, I'm just trying to understand all of this too.
My best guess for the shared parameter is the random intercept `(1 | id)`, but taking a hint from McElreath's code, it doesn't need to be the same in both models (`buy` and `bux`):

```
> mru <- [ulam](https://rdrr.io/pkg/rethinking/man/ulam.html)(
>   [alist](https://rdrr.io/r/base/list.html)(
>     # y model 
>     Y ~ [bernoulli](https://rdrr.io/pkg/brms/man/brmsfamily.html)(p),
>     logit(p) <- a[g] + b_X*X + b_Z*Z[g] + buy*u[g],
>     transpars> vector[Ng]:a <<- abar + z*tau,
>     # X model
>     X ~ normal(mu,sigma),
>     mu <- aX + bux*u[g],
>     vector[Ng]:u ~ normal (0,1),
>     
>     # priors
>     z[g] ~ [dnorm](https://rdrr.io/r/stats/Normal.html)(0,1),
>     [c](https://rdrr.io/r/base/c.html)(aX, b_X, buy, b_Z) ~ [dnorm](https://rdrr.io/r/stats/Normal.html)(0, 1),
>     bux ~ [dexp](https://rdrr.io/r/stats/Exponential.html)(1),
>     abar ~ [dnorm](https://rdrr.io/r/stats/Normal.html) (0,1),
>     tau ~ [dexp](https://rdrr.io/r/stats/Exponential.html)(1),
>     sigma ~ [dexp](https://rdrr.io/r/stats/Exponential.html)(1)
>   ),
>   data = dat, chains = 4, cores=4, sample=TRUE)
```

Anyway, is the stacking necessary any time you have multiple predictors? In your example, that's the case because U and D both need to be centered, which leads to splitting the alpha<sub>i</sub> with indicator variables, but not for the simpler regression of equation 1. In their example, Z is at the group-level, so it doesn't need to be centered.
